content/blog/2018/data-plane-setup/index.md

```diff
@@ -0,0 +1,344 @@
+{{< text yaml>}}
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: ignored
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      containers:
+      - name: ignored
+        image: tutum/curl
+        command: ["/bin/sleep","infinity"]
+
+{{< /text >}}
+
+So it can be seen that there are many variables, based on which automatic sidecar injection is controlled in your pod namespace, and they are:
+
+- webhooks `namespaceSelector` (`istio-injection: enabled`)
+- default policy (Configured in the ConfigMap `istio-sidecar-injector`)
+- per-pod override annotation (`sidecar.istio.io/inject`)
+
+This [table] (<https://istio.io/help/ops/setup/injection/>) shows a clear picture of the final injection status based on the value of the above variables.
+
+## Traffic flow from application container to sidecar proxy
+
+Now that we are clear about how a sidecar container and an init container are injected into an application manifest, how does the sidecar proxy grab the inbound/outbound traffic to/from the container ? We did briefly mention that it is done by setting up the `iptable` rules within the pod namespace, which in turn is done by the `istio-init` container. Now, it is time to verify what actually gets updated within the namespace.
+
+Let’s get into the application pod namespace that we deployed in the previous section and look at the iptables configured. I am going to show an example using `nsenter` (you can also enter the container in a privileged mode to see the same - For folks without access to the nodes, `exec` into the sidecar and running iptables is more practical).
+
+{{< text bash >}}
+$ docker inspect b8de099d3510 --format '{{ .State.Pid }}'
+4125
+{{< /text  >}}
+
+{{< text bash >}}
+$ nsenter -t 4215 -n iptables -t nat -S
+-P PREROUTING ACCEPT
+-P INPUT ACCEPT
+-P OUTPUT ACCEPT
+-P POSTROUTING ACCEPT
+-N ISTIO_INBOUND
+-N ISTIO_IN_REDIRECT
+-N ISTIO_OUTPUT
+-N ISTIO_REDIRECT
+-A PREROUTING -p tcp -j ISTIO_INBOUND
+-A OUTPUT -p tcp -j ISTIO_OUTPUT
+-A ISTIO_INBOUND -p tcp -m tcp --dport 80 -j ISTIO_IN_REDIRECT
+-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15001
+-A ISTIO_OUTPUT ! -d 127.0.0.1/32 -o lo -j ISTIO_REDIRECT
+-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
+-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
+-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
+-A ISTIO_OUTPUT -j ISTIO_REDIRECT
+-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
+{{< /text >}}
+
+From the output above, it is now clear that all the incoming traffic to port 80, which is where our application is listening is now `REDIRECTED` to port `15001`, which is the istio-proxy (envoy) listen port. The same holds true for the outgoing traffic too.
+
+This brings us to the end of this post. I hope it helped to de-mystify how Istio manages to plugin its data-plane components into an existing deployment, mainly injection of the sidecar containers and traffic routing to the proxy.
```

Comment:
Hi Rigs

Was wondering, if you had a look at this and if it looks good to be merged.

Thnx
Manish